protected/controllers/DisorderController.php

```diff
@@ -87,7 +87,7 @@ public function actionCreate()
         if (isset($_POST['Disorder'])) {
             $disorder->attributes = $_POST['Disorder'];
 
-            if ($disorder->save()) {
+            if ($disorder->validate() && $disorder->save()) {
```

Comment:
Does this depend on short-circuiting of the if statement to prevent the '$disorder->save()' being run? It works but it doesn't seem like a clear thing for readability. Also although it's unlikely to change, it seems like a bad thing to rely on when the alternative (a couple of ifs) is so clear.

Reply:
You guys really need to try out some perl one day. in perl you short-circuit all the time, for example
$file = open('filename') or die $!;